core/modules/entity/entity.controller.inc

```diff
@@ -592,49 +599,43 @@ class DefaultEntityController implements EntityControllerInterface {
-
-    $entity->content;
   }
 
   /**
    * Implements EntityControllerInterface::view().
    */
-  public function view(EntityInterface $entity, $view_mode = 'full', $langcode = NULL, $context = array()) {
+  public function view($entities, $view_mode = 'full', $langcode = NULL, $page = NULL) {
+    // @todo ensure entities are keyed by entity id.
     if (!empty($this->entityInfo['fieldable'])) {
-      field_attach_prepare_view($this->entityType, array($entity->id() => $entity), $view_mode);
+      field_attach_prepare_view($this->entityType, $entities, $view_mode);
     }
-    entity_prepare_view($this->entityType, array($entity->id() => $entity));
+    entity_prepare_view($this->entityType, $entities);
     $langcode = isset($langcode) ? $langcode : $GLOBALS['language_content']->langcode;
 
     $view = array();
-
-    $this->buildContent($entity, $view_mode, $langcode, $context);
-
-    $build = $entity->content;
-    // We don't need duplicate rendering info in entity->content.
-    unset($entity->content);
-
-    // If $context['page'] exists it will control how the entity is rendered:
-    // if TRUE the entity will be rendered without its title, so that it can be
-    // embeded in another context. If FALSE the entity will be displayed with
-    // its title in a mode suitable for lists. If unset, the page mode will be
-    // enabled if the current path is the URI of the entity, as returned by
-    // entity_uri().
-    $page = isset($context['page']) ? $context['page'] : NULL;
-
-    $build += array(
-      // If the entity type provides an implementation, use this instead the
-      // generic one.
-      // @see template_preprocess_entity()
-      '#theme' => 'entity',
-      '#entity_type' => $this->entityType,
-      '#entity' => $entity,
-      '#view_mode' => $view_mode,
-      '#language' => $langcode,
-      '#page' => $page,
-    );
-    // Allow modules to modify the structured entity.
-    backdrop_alter(array($this->entityType . '_view', 'entity_view'), $build, $this->entityType);
+    foreach ($entities as $entity) {
+      /* @var EntityInterface $entity */
+      $this->buildContent($entity, $view_mode, $langcode);
+
+      $build = $entity->content;
+      // We don't need duplicate rendering info in entity->content.
+      unset($entity->content);
+
+      $build += array(
+        // If the entity type provides an implementation, use this instead the
+        // generic one.
+        // @see template_preprocess_entity()
+        '#theme' => 'entity',
+        '#entity_type' => $this->entityType,
+        '#entity' => $entity,
+        '#view_mode' => $view_mode,
+        '#language' => $langcode,
+        '#page' => $page,
+      );
+      // Allow modules to modify the structured entity.
+      backdrop_alter(array($this->entityType . '_view', 'entity_view'), $build, $this->entityType);
+      $view[$this->entityType][$entity->id()] = $build;
```

Comment:
The abstract class `Entity` currently does not currently provide a `id()` method, even though its definition it's included in the interface `EntityInterface`.  We could create this method.

Reply:
It's not needed as part of this PR - there are other parts of code that are calling `$entity->id()` too.